Type declarations missing

Hi,
can you add the missing type declarations to the package, please?
```
$ npm install @types/react-native-youtube

npm ERR! code E404
npm ERR! 404 Not Found - GET https://registry.npmjs.org/@types%2freact-native-youtube - Not found
npm ERR! 404 
npm ERR! 404  '@types/react-native-youtube@latest' is not in the npm registry.
npm ERR! 404 You should bug the author to publish it (or use the name yourself!)
npm ERR! 404 
npm ERR! 404 Note that you can also install from a
npm ERR! 404 tarball, folder, http url, or git url.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/sirhennihau/.npm/_logs/2020-03-19T22_56_16_568Z-debug.log
```
Version:
    "react-native-youtube": "^2.0.0",

--- a/main.d.ts
+++ b/main.d.ts
@@ -14,16 +14,25 @@ export interface YouTubeProps {
   showFullscreenButton?: boolean;
   rel?: boolean;
   origin?: string;
-  onError?: () => void;
-  onReady?: () => void;
+  onError?: (event:any) => void;
+  onReady?: (event:any) => void;
   onChangeState?: () => void;
   onChangeQuality?: () => void;
-  onChangeFullscreen?: () => void;
-  onProgress?: () => void;
+  onChangeFullscreen?: (event:any) => void;
+  onProgress?: (event:any) => void;
   style?: StyleProp<ViewStyle>;
 }
 
-declare class YouTube extends React.Component<YouTubeProps> {}
+declare class YouTube extends React.Component<YouTubeProps> {
+  seekTo(seconds:number):void
+  nextVideo():void
+  previousVideo():void
+  playVideoAt(index:number):void
+  getVideosIndex():Promise<number>
+  getCurrentTime():Promise<number>
+  getDuration():Promise<number>
+  reloadIframe():void
+}
 
 export declare const YouTubeStandaloneIOS: {
   playVideo(videoId: string): Promise<void>;

--- a/main.d.ts
+++ b/main.d.ts
@@ -1,0 +1,58 @@
+import * as React from 'react';
+import { StyleProp, ViewStyle } from 'react-native';
+
+export interface YouTubeProps {
+  videoId?: string;
+  videoIds?: string[];
+  playlistId?: string;
+  play?: boolean;
+  loop?: boolean;
+  fullscreen?: boolean;
+  controls?: 1 | 2 | 3;
+  showinfo?: boolean;
+  modestbranding?: boolean;
+  showFullscreenButton?: boolean;
+  rel?: boolean;
+  origin?: string;
+  onError?: () => void;
+  onReady?: () => void;
+  onChangeState?: () => void;
+  onChangeQuality?: () => void;
+  onChangeFullscreen?: () => void;
+  onProgress?: () => void;
+  style?: StyleProp<ViewStyle>;
+}
+
+declare class YouTube extends React.Component<YouTubeProps> {}
+
+export declare const YouTubeStandaloneIOS: {
+  playVideo(videoId: string): Promise<void>;
+};
+
+export declare const YouTubeStandaloneAndroid: {
+  playVideo(params: {
+    apiKey: string;
+    videoId: string;
+    autoplay?: boolean;
+    lightboxMode?: boolean;
+    startTime?: number;
+  }): Promise<void>;
+  playVideos(params: {
+    apiKey: string;
+    videoIds: string[];
+    autoplay?: boolean;
+    lightboxMode?: boolean;
+    startIndex?: number;
+    startTime?: number;
+  }): Promise<void>;
+  playPlaylist(params: {
+    apiKey: string;
+    playlistId: string[];
+    autoplay?: boolean;
+    lightboxMode?: boolean;
+    startIndex?: number;
+    startTime?: number;
+  }): Promise<void>;
+};
+
+export default YouTube;

--- a/package.json
+++ b/package.json
@@ -3,6 +3,7 @@
   "version": "2.0.0",
   "description": "A <YouTube/> component for React Native.",
   "main": "main.js",
+  "types":"main.d.ts",
   "scripts": {
     "test": "echo \"Error: no test specified\" && exit 1"
   },
@@ -28,6 +29,9 @@
     "react-native": ">=0.60"
   },
   "devDependencies": {
-    "prettier": "^1.18.2"
+    "prettier": "^1.18.2",
+    "typescript":"^3.8.3",
+    "@types/react":"^16.9.34",
+    "@types/react-native":"^0.62.2"
   }
 }

--- a/tests/type-test.tsx
+++ b/tests/type-test.tsx
@@ -1,0 +1,33 @@
+import * as React from 'react';
+import YouTube, { YouTubeStandaloneAndroid, YouTubeStandaloneIOS } from '../main';
+
+const YouTubeTest = () => {
+  YouTubeStandaloneAndroid.playVideo({
+    apiKey: 'foo',
+    videoId: 'bar',
+  });
+  YouTubeStandaloneIOS.playVideo('foo');
+  return (
+    <YouTube
+      videoId={'foo'}
+      videoIds={['foo', 'bar']}
+      playlistId={'foo'}
+      play
+      loop
+      fullscreen
+      controls={1}
+      showinfo
+      modestbranding
+      showFullscreenButton
+      rel
+      origin={'foo'}
+      onError={() => {}}
+      onReady={() => {}}
+      onChangeState={() => {}}
+      onChangeQuality={() => {}}
+      onChangeFullscreen={() => {}}
+      onProgress={() => {}}
+      style={{ flex: 1 }}
+    />
+  );
+};

--- a/tests/type-test.tsx
+++ b/tests/type-test.tsx
@@ -7,8 +7,19 @@ const YouTubeTest = () => {
     videoId: 'bar',
   });
   YouTubeStandaloneIOS.playVideo('foo');
+  const videoRef = React.useRef<YouTube>(null)
+  React.useEffect(()=>{
+    videoRef.current.getCurrentTime()
+    videoRef.current.nextVideo()
+    videoRef.current.previousVideo()
+    videoRef.current.reloadIframe()
+    videoRef.current.getDuration()
+    videoRef.current.playVideoAt(1)
+    videoRef.current.seekTo(1000)
+  },[])
   return (
     <YouTube
+      ref={videoRef}
       videoId={'foo'}
       videoIds={['foo', 'bar']}
       playlistId={'foo'}
@@ -21,11 +32,11 @@ const YouTubeTest = () => {
       showFullscreenButton
       rel
       origin={'foo'}
-      onError={() => {}}
-      onReady={() => {}}
+      onError={(event) => {}}
+      onReady={(event) => {}}
       onChangeState={() => {}}
       onChangeQuality={() => {}}
-      onChangeFullscreen={() => {}}
+      onChangeFullscreen={(event) => {}}
       onProgress={() => {}}
       style={{ flex: 1 }}
     />

